Code/Conviva/Source/Events/ConvivaReporters/AdEventConvivaReporter.swift

```diff
@@ -47,7 +47,8 @@ public class AdEventConvivaReporter: AdEventProcessor, ConvivaAdPlaybackEventsRe
         info["c3.csid"] = self.videoAnalytics.getSessionId()
         
         // Temporary workaround for missing LinearAd in Native THEOplayerGoogleIMAIntegration. Can be removed after THEO-10161 is completed.
```

Comment:
The idea here is that if the ad that was passed with the adEvent DOES contain the duration it will be part of the 'info' and we can use the original instead of the workaround where we took the player.duration (whoch does not make sense in an SSAI context)